fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/expressions/Comparisons.java

```diff
@@ -954,18 +1002,32 @@ public int planHash(@Nonnull final PlanHashKind hashKind) {
                 case LEGACY:
                     return type.name().hashCode() + parameter.hashCode();
                 case FOR_CONTINUATION:
-                    return PlanHashable.objectsPlanHash(hashKind, BASE_HASH, type, parameter);
                 case STRUCTURAL_WITHOUT_LITERALS:
-                    return PlanHashable.objectsPlanHash(hashKind, BASE_HASH, type);
+                    return PlanHashable.objectsPlanHash(hashKind, BASE_HASH, type, parameter);
                 default:
                     throw new UnsupportedOperationException("Hash Kind " + hashKind.name() + " is not supported");
             }
         }
 
         @Override
         public int queryHash(@Nonnull final QueryHashKind hashKind) {
-            // Query hash without parameters ignores parameter.
-            return HashUtils.queryHash(hashKind, BASE_HASH, type);
+            return HashUtils.queryHash(hashKind, BASE_HASH, type, parameter);
```

Comment:
Since we now actually insist on parameter marker names being the same I changed the implementation for query hash to also include the parameter name regardless if `LITERALS` or `NO_LITERALS` is used. Parameters are strictly speaking not literals but identifiers defined by the user. In any case we depend on these markers being consistent and somewhat stable and I think that is reflected here.